ui/src/hooks/useProofRequest.ts

```diff
@@ -57,11 +53,11 @@ export const useProofRequest = () => {
           }),
         )
       } catch (error) {
-        setIsGeneratingMerkleProof(false)
-        setIsSubmitError({
-          error,
-          isError: true,
+        setGenerateMerkleProofError(() => {
```

Comment:
This is the only part that is a bit weird.
See
https://github.com/vercel/next.js/discussions/50564